decentdao/decent-interface
# `1931` Part 1: Local storage updates

Fixes #1931 

Makes updates Throughout app. This is for updating from the long string format for the storage key. to an "Object" key. While this is ultimately still a `string` this will allow for much better readability and adds versioning to each key (asides from having to update a few types as well as type safety while getting and setting values. 

This also will add `cross-network` favorites combining how we use local storage for caching users favorites to allow for navigating to different networks

## Merged PRs
- https://github.com/decentdao/decent-interface/pull/1975
- https://github.com/decentdao/decent-interface/pull/1984
- https://github.com/decentdao/decent-interface/pull/1996


## For testing
### Testing migration
Time to go into Dev tools a little. 

Here is my old values for local storage.

```
fract_1_favorites: {"v":["0x36bD3044ab68f600f6d3e081056F34f2a58432c4","0xD26c85D435F02DaB8B220cd4D2d398f6f646e235"],"e":-1}
fract_11155111_favorites: {"v":["0xDD73b64C8845868CCddB30df83929fBA63b4d284","0x76DA0508ee8f900be06E58D8D56F418494C266d7","0x0503d9A46BedCA2f9882413b0e10D29d6Daf9C5c","0x259a5770d7242b5aBa7420926690E90c69037d10","0x70484d88cB248b9a5150d133fEDBa7948AfC9b0B","0x1F0A855BCE550A143D5000181f6187bd6F32C612","0x789eD59E7644f5A8ea72206ae1512312cde0B794","0x44Ec57b5FFe102Ed5611DA7FdC632FbF2f8E7Fe9","0xCA491BC3e97aF1E29464Ef426f908239a270BBB9","0xCB38436237d52468B8E319f8729d3a2CF64a6066","0xfe0B8e7F7D4bA35C8416421566D95B19b9AbD092","0xDFF1882cE9Bfb7EF4E73d8C80bB3B7918cAB3d79","0xaD13C0AdD763bAbEc42673a0488511e91b2c843f","0x0516e7A7eC4B4a417f08103Bc38410B3FdAea2BC","0x83276D238A33d373Fb8d22Db6AC5456200AC8Db0","0xD63a1d7bD0A82B8a8032479215EAeBd55FeC3Fc9","0x35b798b19F70b236b9Dcb156F21F4f0f4Ca44C49","0xafEA17d5a9025fED84F52fa0f8f2576436968383","0x38AB2b9BffdDecEcc3795d516393b24E5394D237","0xc235aCFc002094D8C2156e061F040E42BD3bDA91","0xf5175a81140f4d6950870C04DCebBC3974C95C9D","0xd418E98a11B9189fCc05cddfbB10F4Cee996C749","0x2b52aC156d229Ca754c00A1d9042C88Baf191dfd"],"e":-1}
```

- Go into the dev tools in your browser -> Application -> Local Storage -> then look for this site's data (deploy preview)
- Add these key/value pairs into local storage
- Remove Migration key/value pair
- Refresh.

Should create the new favorites

### For Favorites
- Ensure Favorites toggling still works as expected
- Ensure Favorites Navigation still works as expected
- Ensure Favorites Menu/Drawer/MySafes all render correct as design
- (Note) Will not show cross-network DAO Names just yet. Will need @adamgall 's viem updates to make this work smoothly
